songpal/device.py

```diff
@@ -516,3 +528,259 @@ async def raw_command(self, service: str, method: str, params: Any):
         """
         _LOGGER.info("Calling %s.%s(%s)", service, method, params)
         return await self.services[service][method](params)
+
+
+class UpnpDevice(Device):
```

Comment:
Please separate this class into its own file.

Reply:
tried that, but gives some circular dependency issues. would require everything to be even more split up (device, device factory and upnp device) as the __init__.py loads device.py
I find it more concise to have everything in the device.py (as Upnp is just a specalization of device)